packages/apify/src/proxy_configuration.ts

```diff
@@ -339,17 +360,33 @@ export class ProxyConfiguration extends CoreProxyConfiguration {
      * If the check can not be made, it only prints a warning and allows the program to continue. This is to
      * prevent program crashes caused by short downtimes of Proxy.
      */
-    protected async _checkAccess(): Promise<void> {
+    protected async _checkAccess(): Promise<boolean> {
         const status = await this._fetchStatus();
-        if (status) {
-            const { connected, connectionError, isManInTheMiddle } = status;
-            this.isManInTheMiddle = isManInTheMiddle;
 
-            if (!connected) this._throwApifyProxyConnectionError(connectionError);
-        } else {
+        if (!status) {
             this.log.warning('Apify Proxy access check timed out. Watch out for errors with status code 407. '
                 + 'If you see some, it most likely means you don\'t have access to either all or some of the proxies you\'re trying to use.');
+            return true;
         }
+
+        const { connected, connectionError, isManInTheMiddle } = status;
+        this.isManInTheMiddle = isManInTheMiddle;
+
+        if (connected) {
+            return true;
+        }
+
+        // Throw only on the platform, locally we just print a warning and run requests without the proxy.
+        // This is because the user might not have set up things correctly yet.
+        // It still fails on the platform, where we don't want to allow this behavior.
+        if (Actor.isAtHome()) {
+            this._throwApifyProxyConnectionError(connectionError);
+        } else {
+            this.log.warning(connectionError);
+            return false;
```

Comment:
```suggestion
```

I still feel like there are way too many `return`s here... this one we can kill for sure though :)